src/EditorFeatures/Core/Organizing/OrganizeDocumentCommandHandler.cs

```diff
@@ -149,7 +149,7 @@ private void SortAndRemoveUnusedImports(ITextBuffer subjectBuffer, IUIThreadOper
                 operationContext, _threadingContext);
             if (document != null)
             {
-                var formattingOptions = SyntaxFormattingOptions.FromDocumentAsync(document, cancellationToken).WaitAndGetResult(cancellationToken);
+                var formattingOptions = document.SupportsSyntaxTree ? SyntaxFormattingOptions.FromDocumentAsync(document, cancellationToken).WaitAndGetResult(cancellationToken) : null;
```

Comment:
Should `SyntaxFormattingOptions.FromDocumentAsync` handle `document.SupportsSyntaxTree` and return null? If it always expects callers to make this check before invoking the API, can we add an assert inside `SyntaxFormattingOptions.FromDocumentAsync`?

Reply:
Most callers do expect formatting options. SyntaxFormattingOptions indicates we have a syntax tree already. An assertion is not necessary since the method calls GetRequiredService<ISyntaxFormattingService> which will throw if the formatting service is not available (the same as !SupportsSyntaxTree).